syntax = "proto3";

package raft;

service RaftService {
    rpc RequestVote (VoteRequest) returns (VoteResponse);
    rpc AppendEntries (LogEntry) returns (AppendResponse);
}

message VoteRequest {
    string node_id = 1;
    int32 term = 2;
}

message VoteResponse {
    bool vote_granted = 1;
}

message LogEntry {
    int32 term = 1;
    string command = 2;
}

message AppendResponse {
    bool success = 1;
}
